CPAC/nuisance/utils/compcor.py

```diff
@@ -181,9 +181,34 @@ def fallback_svd(a, full_matrices=True, compute_uv=True):
 
 
 def TR_string_to_float(tr):
-    if 'ms' in tr:
-        tr = float(tr.replace('ms', '')) / 1000
-    else:
-        tr = float(tr.replace('s', ''))
+    """
+    Convert TR string to seconds (float). Suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Parameters
+    ----------
+    tr : TR string representation. May use suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Returns
+    -------
+    tr in seconds (float)
+    """
+    if not isinstance(tr, str):
+        raise TypeError(f'Improper type for TR_string_to_float ({tr}).')
+
+    tr_str = tr.replace(' ', '')
+    factor = 1.
+
+    if tr_str.endswith('ms'):
+        tr_str = tr_str[:-2]
+        factor = 0.001
+    elif tr.endswith('s'):
+        tr_str = tr_str[:-1]
+
+    try:
+        tr_numeric = float(tr_str)
+    except Exception as exc:
+        raise ValueError(f'Can not convert TR string to float: "{tr}".') from exc
 
-    return tr
+    return tr_numeric * factor
```

Comment:
Is https://github.com/FCP-INDI/C-PAC/blob/44e2431785cdb953fd8aa53c8767f53c1dfeb810/CPAC/nuisance/utils/compcor.py#L200-L214 way more efficient than
```Python
try:
    if 'ms' in tr:
        tr = float(tr.replace('ms', '')) / 1000
    else:
        tr = float(tr.replace('s', ''))
except Exception as exc:
    raise ValueError(f'Can not convert TR string to float: "{tr}".') from exc
return tr
```
? I personally think the if/else with oneliners is easier to tell what it's doing, and it's fewer lines.

Reply:
I do think `.endswith('ms')` is preferable to `'ms' in`

Reply:
It just felt weird to me to not fail on e.g. `'10ms1'`

Reply:
Yeah, that would fail in a very hard-to-find way, because it would turn into `101/1000` when they probably mean `10/1000`

Reply:
I have broadened the try block and condensed the code a bit in a09d212